src/UnitySampleProject/Assets/Scripts/Prefabs/Gelato/GelatoModule.cs

```diff
@@ -0,0 +1,209 @@
+using System.Numerics;
+using System.Threading;
+using System.Threading.Tasks;
+using ChainSafe.GamingSDK.EVM.MetaMaskBrowserWallet;
+using ChainSafe.GamingWeb3.Build;
+using ChainSafe.GamingWeb3.Unity;
+using ChainSafe.GamingSdk.Gelato;
+using ChainSafe.GamingSdk.Gelato.Dto;
+using ChainSafe.GamingWeb3;
+using UnityEngine;
+using Web3Unity.Scripts.Library.Ethers.JsonRpc;
+using Contract = Web3Unity.Scripts.Library.Ethers.Contracts.Contract;
+
+public class GelatoModule : MonoBehaviour
+{
+    public async void CallWithSyncFeeExample()
+    {
+        var vitalik = "0xd8da6bf26964af9d7eed9e03e53415d37aa96045";
+        var target = "0xA045eb75e78f4988d42c3cd201365bDD5D76D406";
+        var feeToken = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
+        var abi = "[{\"inputs\": [" +
+                  "{\"internalType\":\"address\",\"name\":\"_token\",\"type\":\"address\"}," +
+                  "{\"internalType\":\"address\",\"name\":\"_to\",\"type\":\"address\"}," +
+                  "{\"internalType\":\"uint256\",\"name\":\"_amount\",\"type\":\"uint256\"}" +
+                  "]," +
+                  "\"name\":\"sendToFriend\"," +
+                  "\"outputs\":[]," +
+                  "\"stateMutability\":\"nonpayable\"," +
+                  "\"type\":\"function\"" +
+                  "}]";
+        var contract = Web3Accessor.Web3.ContractBuilder.Build(abi, target);
+        var data = contract.Calldata("sendToFriend", new object[]
+        {
+            feeToken,
+            vitalik,
+            new BigInteger(5 * 10E12),
+        });
+
+        var relayResponse = await Web3Accessor.Web3.Gelato().CallWithSyncFee(new CallWithSyncFeeRequest()
+        {
+            Data = data,
+            FeeToken = feeToken,
+            IsRelayContext = true,
+            Target = target,
+        });
+
+        var complete = false;
+        while (!complete)
+        {
+            var status = await Web3Accessor.Web3.Gelato().GetTaskStatus(relayResponse.TaskId);
+            Debug.Log($"CallWithSyncFee status: {relayResponse.TaskId}: {status.TaskState}");
+
+            switch (status.TaskState)
+            {
+                case TaskState.ExecSuccess:
+                case TaskState.ExecReverted:
+                case TaskState.Cancelled:
+                    Debug.Log("Task complete");
+                    complete = true;
+                    Debug.Log($"Final status of {relayResponse.TaskId}: {status.TaskState}");
+                    Debug.Log($"Transaction hash: {status.TransactionHash}: ");
+                    break;
+                default:
+                    Thread.Sleep(2000);
```

Comment:
Don't use Threads inside tasks as it's a low-level system and will probably just freeze the main thread for 2 seconds in this case. Use `await Task.Delay(TimeSpan.FromSeconds(2));` instead.